fn/list_test.go

```diff
@@ -727,3 +729,93 @@ func TestMoveUnknownMark(t *testing.T) {
 	checkList(t, &l1, []int{1})
 	checkList(t, &l2, []int{2})
 }
+
+// TestFilterIdempotence ensures that the slice coming out of List.Filter is
+// the same as that slice filtered again by the same predicate.
+func TestFilterIdempotence(t *testing.T) {
+	err := quick.Check(
```

Comment:
Nice! I have a PR that adds `rapid` as a dep (https://github.com/lightningnetwork/lnd/pull/9032), it's only in the aux chan staging branch rn though. Once pulled in, we can start to switch over to it, as it's _miles_ ahead of `quick.Check`.